pkg/kubelet/rkt/sync.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rkt
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kubecontainer "github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/container"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/probe"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/volume"
+	"github.com/golang/glog"
+)
+
+// SyncPod makes
+func (r *Runtime) SyncPod(runningPod kubecontainer.Pod, pod *api.Pod, podStatus api.PodStatus, podVolumes map[string]volume.Volume) error {
+	podFullName := kubecontainer.GetPodFullName(pod)
+	uid := pod.UID
+
+	if len(runningPod.Containers) == 0 {
+		glog.V(4).Infof("Pod is not running, start it")
+		return r.RunPod(pod, podVolumes)
+	}
+
+	// Add references to all containers.
+	unidentifiedContainers := make(map[types.UID]*kubecontainer.Container)
+	for _, c := range runningPod.Containers {
+		unidentifiedContainers[c.ID] = c
+	}
+
+	restartPod := false
+	for index, container := range pod.Spec.Containers {
+		expectedHash := hashContainer(&container)
+
+		c := runningPod.FindContainerByName(container.Name)
+		if c == nil {
+			if shouldContainerBeRestarted(&container, pod, podStatus, r.readinessManager) {
+				glog.V(3).Infof("Container %+v is dead, but RestartPolicy says that we should restart it.", container)
+				// TODO(yifan): Containers in one pod are fate-sharing at this moment, see:
+				// https://github.com/appc/spec/issues/276.
+				restartPod = true
+				break
+			}
+			continue
+		}
+
+		containerChanged := c.Hash != 0 && c.Hash != expectedHash
+		if containerChanged {
+			glog.Infof("Pod %q container %q hash changed (%d vs %d), it will be killed and re-created.", podFullName, container.Name, c.Hash, expectedHash)
+			restartPod = true
+			break
+		}
+
+		result, err := r.prober.Probe(pod, podStatus, container, string(c.ID), c.Created)
+		if err != nil {
+			// TODO(vmarmol): examine this logic.
+			glog.V(2).Infof("Probe no-error: %q", container.Name)
+			delete(unidentifiedContainers, c.ID)
+			continue
+		}
+		if result == probe.Success {
+			glog.V(4).Infof("Probe success: %q", container.Name)
+			delete(unidentifiedContainers, c.ID)
+			continue
+		}
+		glog.Infof("Pod %q container %q is unhealthy (probe result: %v), it will be killed and re-created.", podFullName, container.Name, result)
+		restartPod = true
```

Comment:
shall we also break out in this case? Actually i think we should follow the previous logic here. If ...{ restartPod = true; break}; normal logic